src/posts/2021-01-20-ebpf-updates/index.md

```diff
@@ -0,0 +1,314 @@
+  additional information to uniquely identify shared mappings, see
+  [`perf_event_open` man page](https://www.man7.org/linux/man-pages/man2/perf_event_open.2.html)).
+  (Jiri Olsa,
+  [link](https://lore.kernel.org/bpf/20210114134044.1418404-1-jolsa@kernel.org/t/#u))
+
+* Support retrieval of more `SOL_SOCKET` level options from `sock_addr` eBPF
+  programs, to fill the gap between the list of options that `bpf_setsockopt()`
+  can set and those that `bpf_getsockopt()` could retrieve. The concerned
+  options are:
+  * `SO_MARK`
+  * `SO_PRIORITY`
+  * `SO_BINDTOIFINDEX` (also new for `bpf_setsockopt()`)
+
+  (Daniel Borkmann,
+  [link](https://lore.kernel.org/bpf/cba44439b801e5ddc1170e5be787f4dc93a2d7f9.1610406333.git.daniel@iogearbox.net/t/#u))
+
+* Improve out-of-tree cross-building for eBPF selftests. Although this adds no
+  new feature, it feels interesting to report because it should enable wider
+  automated testing on ARM architectures. Selftests are, of course, an
+  essential part of the eBPF ecosystem.
+  (Jean-Philippe Brucker,
+  [link](https://lore.kernel.org/bpf/20210113163319.1516382-1-jean-philippe@linaro.org/t/#u))
+
+## Did You Know?: eBPF Virtual Filesystem
+
+eBPF objects, such as a program or a map, reside in kernel memory until they
+are no longer needed. Internally, the kernel uses reference counters to keep
+track of the number of “handles” pointing to such objects. When the number of
+references comes down to zero, the program or the map is destroyed. The
+references to a program would typically be a hook where the user has attached
+the program (such as a tc filter or a kernel probe), or file descriptors to the
+program that were returned from the kernel when loading the program with the
+`bpf()` system call. Similarly, references to an eBPF map can be held by eBPF
+programs using the map or by a user program that retrieved a file descriptor.
+
+This means that if a process loaded an eBPF program but has not attached it,
+the program will be destroyed when the process exists and its file descriptors
+are closed. There are ways to share file descriptors between processes, but to
+make it easier to reference eBPF objects between user applications, or simply
+to make them persistent at a time when they have no reference in the kernel
+(such as a detached program or an unused map), another mechanism has been
+created: the eBPF virtual filesystem.
+
+The eBPF virtual (or _pseudo_) filesystem, often called bpffs, is traditionally
+mounted at `/sys/fs/bpf`, but any alternative mount point can work. It is
+possible to pin objects to this virtual filesystem, which is rendered as file
+paths. Calling the `bpf()` system call with its `BPF_OBJ_PIN` subcommand pins
+an object, using the `BPF_OBJ_GET` on a bpffs path retrieves a file descriptor
+to the pinned object. Removing a pinned path simply involves a call to
+`unlink()`, just like for regular paths. They are not persistent after reboot.
```

Comment:
It's a bit hard to grasp what "They" refers to here. Maybe say "eBPF objects" explicitly?

```suggestion
`unlink()`, just like for regular paths. eBPF objects are not persistent after reboot.
```

Reply:
I was in fact thinking of the pinned paths :). Now clarified, thanks.

Reply:
I see 😄  thanks for clarifying.